### ISSUE TITLE
LOOPS: tests broken when Semigroups are loaded

### ISSUE THREAD

**Issue by alex-konovalov** [gap-packages/Semigroups#361]
The standard test of the LOOPS package pass when GAP is started with no packages or with default packages, but fails when all packages are loaded. 
```
########> Diff in /Users/alexk/gap4r8p8/pkg/loops/tst/core_methods.tst, line 
222:
# Input is:
AllSubquasigroups( QuasigroupByCayleyTable( [[2,1],[1,2]] ) );
# Expected output:
[ <quasigroup of order 2>, <quasigroup of order 1> ]

# But found:
Error, Usage: Semigroup(<gen>,...), Semigroup(<gens>), Semigroup(<D>),
########
```
One should be able to reproduce this by running

    make testpackage PKGNAME=loops

and inspecting log files in `dev/log` (I've observed this in GAP 4.8.8, which I will make available shortly).

My first guess was to look at Semigroups, and indeed, the backtrace points to its code:
```
gap> AllSubquasigroups( QuasigroupByCayleyTable( [[2,1],[1,2]] ) );
Error, Usage: Semigroup(<gen>,...), Semigroup(<gens>), Semigroup(<D>), called from
Semigroup( NextIterator( iter ) 
 ) at /Users/alexk/gap4r8p8/pkg/semigroups-2.8.0/gap/semitrans.gi:100 called from
GeneratorsSmallest( b ) at /Users/alexk/gap4r8p8/pkg/loops/gap/core_methods.gi:100 called from
Set( res ) at /Users/alexk/gap4r8p8/lib/list.gi:708 called from
Set( Elements( Q ), function ( x )
      return Subquasigroup( Q, [ x ] );
  end ) at /Users/alexk/gap4r8p8/pkg/loops/gap/core_methods.gi:552 called from
<function "unknown">( <arguments> )
 called from read-eval loop at line 5 of *stdin*
type 'quit;' to quit to outer loop
brk> 
```

**Comment by james-d-mitchell** [comment]
Ok will investigate

**Comment by james-d-mitchell** [comment]
The last release version of Semigroups 2.8 was 2.8.1, is there some reason you're not using that? Also this was released over a year ago, is it possible that this is an issue more with loops than with Semigroups since nothing changed for over a year? Or is the running of this test new?

**Comment by james-d-mitchell** [comment]
@alexkonovalov?

**Comment by james-d-mitchell** [comment]
@alex-konovalov

**Comment by alex-konovalov** [comment]
@james-d-mitchell yes, your 2nd guess about my username was correct ;)

**Comment by alex-konovalov** [comment]
@james-d-mitchell The problem is not new -  I can reproduce it in GAP 4.8.7 too. Just discovered during tests cleanup. Usually, for packages that have non-reproducible tests, we recommend that at least the test passes with default packages, and for LOOPS this is the case - that's why it was not reported earlier. 

About semigroups 2.8.1 - it has never been picked up by the package update mechanism. The logs from the package update system show that 2.8.0 has been replaced by 3.0.1 (see below the history of updates discovered and imported by the GAP package update system).

Perhaps 2.8.1 archive has never been made discoverable by the GAP package update system? Where can I get it?

```
changeset:   47:d5049a29f589
tag:         tip
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Thu Aug 24 09:02:00 2017 +0100
summary:     Added tag Version-3.0.5 for changeset dced618080bd

changeset:   46:dced618080bd
tag:         Version-3.0.5
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Thu Aug 24 09:02:00 2017 +0100
summary:     Version 3.0.5, 23/08/2017

changeset:   45:f34e36bbb137
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jul 17 09:02:04 2017 +0100
summary:     Added tag Version-3.0.4 for changeset c1e6ccbd69b3

changeset:   44:c1e6ccbd69b3
tag:         Version-3.0.4
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jul 17 09:02:04 2017 +0100
summary:     Version 3.0.4, 16/07/2017

changeset:   43:8c2a40990954
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Thu Jun 22 09:01:44 2017 +0100
summary:     Added tag Version-3.0.3 for changeset 5c17a3406ac2

changeset:   42:5c17a3406ac2
tag:         Version-3.0.3
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Thu Jun 22 09:01:44 2017 +0100
summary:     Version 3.0.3, 21/06/2017

changeset:   41:640d6bb3c49e
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jun 19 09:18:01 2017 +0100
summary:     Added tag Version-3.0.2 for changeset b70a0bc72c0d

changeset:   40:b70a0bc72c0d
tag:         Version-3.0.2
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jun 19 09:18:00 2017 +0100
summary:     Version 3.0.2, 16/06/2017

changeset:   39:cc7ad2e569b4
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jun 05 09:03:09 2017 +0100
summary:     Added tag Version-3.0.1 for changeset e2d18a7ff254

changeset:   38:e2d18a7ff254
tag:         Version-3.0.1
user:        gap-jenkins@kovacs.cs.st-andrews.ac.uk
date:        Mon Jun 05 09:03:09 2017 +0100
summary:     Version 3.0.1, 03/06/2017

changeset:   37:bb80cd3758cc
user:        gap <gap@mcs.st-andrews.ac.uk>
date:        Thu May 26 13:54:29 2016 +0100
summary:     Added tag Version-2.8.0 for changeset 44414dbe55e2

changeset:   36:44414dbe55e2
bookmark:    gap4r8p4_2016_06_04-12_41
bookmark:    gap4r8p5_2016_09_25-11_49
bookmark:    gap4r8p6_2016_11_12-14_25
bookmark:    gap4r8p7_2017_03_24-21_21
bookmark:    stable
tag:         Version-2.8.0
user:        gap <gap@mcs.st-andrews.ac.uk>
date:        Thu May 26 13:54:28 2016 +0100
summary:     Version 2.8.0, 26/05/2016

changeset:   35:e5c02a5a3e5f
user:        gap <gap@mcs.st-andrews.ac.uk>
date:        Tue Apr 19 15:10:39 2016 +0100
summary:     Added tag Version-2.7.6 for changeset 654919e29fb5

changeset:   34:654919e29fb5
tag:         Version-2.7.6
user:        gap <gap@mcs.st-andrews.ac.uk>
date:        Tue Apr 19 15:10:39 2016 +0100
summary:     Version 2.7.6, 19/04/2016
```

**Comment by james-d-mitchell** [comment]
This is fixed in Version 3.0.5, I'm reluctant to make another release of 2.8.x, given that it is now deprecated. Will it cause any problems in releasing GAP 4.8.8 to not fix this issue @alex-konovalov ?

**Comment by alex-konovalov** [comment]
Thanks @james-d-mitchell! I would be happy if the fix is only for GAP 4.9 release series. I think we may close this issue then.

**Comment by james-d-mitchell** [comment]
Great, will close.
